reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py

```diff
@@ -122,9 +122,7 @@ def _assemble_subscribers_with_auto_promotions(self) -> None:
                         blocked_versions=blocked_versions.get(
                             resource_template.url, set()
                         ),
-                        # Note: this will be refactored at a later point.
-                        # https://issues.redhat.com/browse/APPSRE-7516
-                        use_target_config_hash=bool(saas_file.publish_job_logs),
+                        use_target_config_hash=bool(saas_file.has_test_jobs),
```

Comment:
Is this the only usage of `has_test_jobs`? I don't know what features are different when `has_test_jobs` set to `True` or `False`.

Reply:
Currently usage of target_config_hash is the only difference

Reply:
When to set it to `True` or `False`? As a user, what can I get by setting it?

Reply:
If you set it you will have target_config_hash set in promotion_data

Reply:
This is existing functionality - nothing changes about that. This is just a more declarative flag, to remove hidden meaning of publishJobLogs

Reply:
More details are in the ticket https://issues.redhat.com/browse/APPSRE-7516

Reply:
Treat me like a first time user, I don't know anything about `publishJobLogs` or `has_test_jobs`, I guess by setting `publishJobLogs`, I can get the job logs somewhere, but I can't image what can I get by setting `has_test_jobs`. Do we really need it?

Reply:
This is described in detail in the ticket already

Reply:
If you have an alternate suggestion, please tell me